independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java

```diff
@@ -155,33 +153,40 @@ Template parse() {
                     // Flush the last text segment
                     flushText();
                 } else {
-                    String reason;
-                    ErrorCode code;
+                    String reason = null;
+                    ErrorCode code = null;
                     if (state == State.TAG_INSIDE_STRING_LITERAL) {
                         reason = "unterminated string literal";
                         code = ParserError.UNTERMINATED_STRING_LITERAL;
                     } else if (state == State.TAG_INSIDE) {
-                        reason = "unterminated section";
-                        code = ParserError.UNTERMINATED_SECTION;
+                        // First handle the optional end tags and if an unterminated section is found the then throw an exception
```

Comment:
I suppose _found the then throw_ is a typo? But it's not worth re-running CI for that...

Reply:
You're right :facepalm: